### PR TITLE
Watching for the Secrets resources are not owned by the Infinispan CR

### LINKED ISSUE #1206: Watching for the Secrets resources are not owned by the Infinispan CR
There are possible to use custom created resources, i.e. Secrets which are not owned by the Infinispan CR.
Currently, the custom defined Secrets are updated with Infinispan CR as a Controller OwnerReference. This solution works fine for the triggering changes but will cause an unexpected object deletion when Infinispan CR has deleted.

We need to implement `EnqueueRequestsFromMapFunc` for the `Secret` which are not autogenerated and don't have Infinispan CR as owner reference.
https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.4/pkg/handler/enqueue_mapped.go

Probably, we need to migrate owner references removing it from the Secret. Need to investigate more.
Secret names should be stored in the labels also for the indexing get list of the Infinispan CR's or we need to add required fields to the index for caching.

### PR DESCRIPTION
**Pull request by dmvolod**
fixes #1206 

- [x]  TODO: Migration need to be tested

### CODE REVIEW

File: pkg/controller/infinispan/resources/secret/secret_controller.go [hunk 1]
```diff
@@ -81,17 +81,13 @@ func Add(mgr manager.Manager) error {
 
 func (s *secretResource) Process() (reconcile.Result, error) {
 	// Reconcile Encryption Secrets
-	if result, err := s.reconcileKeystoreSecret(); result != nil {
-		return *result, err
-	}
-
```
Comment (rigazilla) [inline pkg/controller/infinispan/resources/secret/secret_controller.go:87]: it's not clear to me why we don't need this any more

**Review by rigazilla — COMMENTED** [review]
added some questions

File: pkg/controller/infinispan/resources/secret/secret_controller.go (hunk 1 above)
Comment (dmvolod) [reply pkg/controller/infinispan/resources/secret/secret_controller.go:87]: This function just set owner reference in the code, nothing more

File: pkg/controller/infinispan/infinispan_controller.go [hunk 2]
```diff
@@ -704,6 +755,21 @@ func (r *ReconcileInfinispan) upgradeInfinispan(infinispan *infinispanv1.Infinis
...
 }
 
+func (r *ReconcileInfinispan) dropSecretOwnerReference(secretName string, infinispan *infinispanv1.Infinispan, logger logr.Logger) error {
+	secret := &corev1.Secret{}
+	if result, err := kube.LookupResource(secretName, infinispan.Namespace, secret, r.client, logger, eventRec); result != nil {
+		return err
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, secret, func() error {
+		creationTimestamp := secret.GetCreationTimestamp()
+		if !creationTimestamp.IsZero() {
```
[the remark this answers is not in the document]
Comment (dmvolod) [reply pkg/controller/infinispan/infinispan_controller.go:765]: The identification passed before in the upper function. This only update

File: pkg/controller/infinispan/infinispan_controller.go [hunk 3]
```diff
@@ -704,6 +755,21 @@ func (r *ReconcileInfinispan) upgradeInfinispan(infinispan *infinispanv1.Infinis
 	})
 }
 
+func (r *ReconcileInfinispan) dropSecretOwnerReference(secretName string, infinispan *infinispanv1.Infinispan, logger logr.Logger) error {
+	secret := &corev1.Secret{}
+	if result, err := kube.LookupResource(secretName, infinispan.Namespace, secret, r.client, logger, eventRec); result != nil {
+		return err
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, secret, func() error {
```
Comment (ryanemerson) [inline pkg/controller/infinispan/infinispan_controller.go:763]: Do we need todo `LookupResource` and `CreateOrUpdate`? Can't we just use `CreateOrPatch`?

The value of `result` is never used, as eventually `destroyResources` always returns `reconcile.Result{}, err` if an error exists.

Comment (dmvolod) [reply pkg/controller/infinispan/infinispan_controller.go:763]: Yes, good point for optimization. Thanks, @ryanemerson

File: pkg/controller/infinispan/resources/secret/secret_controller.go (hunk 1 above)
Comment (rigazilla) [reply pkg/controller/infinispan/resources/secret/secret_controller.go:87]: oh keystore is never generated, right?

Comment (dmvolod) [reply pkg/controller/infinispan/resources/secret/secret_controller.go:87]: Seems to yes

**Review by rigazilla — APPROVED** [review]

**Comment by ryanemerson** [comment]
Thanks @dmvolod

**Pull request merged**